core/src/main/c/windows/net.c

```diff
@@ -166,9 +193,22 @@ JNIEXPORT jboolean JNICALL Java_io_questdb_network_Net_bindUdp
     return FALSE;
 }
 
-JNIEXPORT jlong JNICALL Java_io_questdb_network_Net_connect
+JNIEXPORT jint JNICALL Java_io_questdb_network_Net_connect
         (JNIEnv *e, jclass cl, jlong fd, jlong sockAddr) {
-    jlong res = connect((SOCKET) fd, (const struct sockaddr *) sockAddr, sizeof(struct sockaddr));
+    jint res = connect((SOCKET) fd, (const struct sockaddr *) sockAddr, sizeof(struct sockaddr));
```

Comment:
I haven't spotted code setting `CLOEXEC`. Without it, if a library forks anywhere (e.g., by other code, when the database is embedded), the socket will be inherited by the child causing errors that are quite tricky to debug.
Also consider setting `SOCK_NOSIGPIPE` on MacOS to have socket functions error instead of having a signal handler called.

Reply:
is there a practical scenario, where `CLOEXEC` can be useful or in fact critical in context of QuestDB Java Client ?

Reply:
When QuestDB is used as a library, e.g. in an embedded context or as an ILP client library) we have no control of what other code is up to.
If such other code creates child processes the file descriptor may be leaked. In such case, closing the socket in the parent is not enough to release bound port itself, as the child still holds it.
For example, if the database restarts without killing the child first, without cloexec the server will not be able bind to its original listening port.
This is less of an issue when it's used as a client because the outbound port is chosen dynamically, but it is nevertheless a file descriptor leak.
Whilst not necessarily common, file descriptor leak issues can be quite tricky and time-consuming to track down.